compiler/rustc_codegen_cranelift/src/driver/aot.rs

```diff
@@ -68,14 +68,13 @@ fn reuse_workproduct_for_cgu(
     cgu: &CodegenUnit<'_>,
     work_products: &mut FxHashMap<WorkProductId, WorkProduct>,
 ) -> CompiledModule {
-    let incr_comp_session_dir = tcx.sess.incr_comp_session_dir();
```

Comment:
Is something keeping the handling of the incr-comp session directory from being removed from Session?

Reply:
It has to be stored somewhere. `TyCtxt` won't easily work as cg_ssa and cg_llvm require `sess.incr_comp_session_dir()` at a place where `TyCtxt` is not available. It is probably possible to refactor this away from `Session` and I was already looking into it a bit, but then I got distracted by refactorings to the crate loader.